hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/QueuePath.java

```diff
@@ -61,12 +61,13 @@ public QueuePath(String fullPath) {
   }
 
   /**
-   * Concatenate queue path parts into one queue path string.
-   * @param parts Parts of the full queue pathAutoCreatedQueueTemplate
-   * @return full path of the given queue parts
+   * Constructor to create Queue path from queue names.
+   * The provided queue names will be concatenated by dots, giving a full queue path.
+   * @param queues Queue names
+   * @return QueuePath object
    */
-  public static String concatenatePath(String... parts) {
-    return String.join(DOT, parts);
+  public static QueuePath createFromQueues(String... queues) {
```

Comment:
This is not necessarily an improvement, the strings are in fact parts of the queue path, which are only short queue names, however those are ambiguous, so calling them "queue" might be a bit misleading, since nothing guarantees you can query these queues.

Reply:
hmm, you are right. Renamed it back to the original version. THanks